client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/SubmissionEditForm.jsx

```diff
@@ -343,116 +347,96 @@ class SubmissionEditForm extends Component {
       );
     }
     return null;
-  }
+  };
 
-  renderQuestionGrading(id) {
-    const { attempting, published, graderView } = this.props;
+  const renderQuestionGrading = (id) => {
     const editable = !attempting && graderView;
     const visible = editable || published;
 
     return visible ? <QuestionGrade id={id} editable={editable} /> : null;
-  }
+  };
 
-  renderQuestions() {
-    const {
-      intl,
-      attempting,
-      questionIds,
-      questions,
-      historyQuestions,
-      topics,
-      graderView,
-      showMcqMrqSolution,
-      handleToggleViewHistoryMode,
-      questionsFlags,
-    } = this.props;
-    return (
-      <>
-        {questionIds.map((id, index) => {
-          const question = questions[id];
-          const { answerId, topicId, viewHistory } = question;
-          const topic = topics[topicId];
-          return (
-            <Element
-              name={`step${index}`}
-              key={id}
-              style={styles.questionContainer}
+  const renderQuestions = () => (
+    <>
+      {questionIds.map((id, index) => {
+        const question = questions[id];
+        const { answerId, topicId, viewHistory } = question;
+        const topic = topics[topicId];
+        return (
+          <Element
+            name={`step${index}`}
+            key={id}
+            style={styles.questionContainer}
+          >
+            <SubmissionAnswer
+              {...{
+                readOnly: !attempting,
+                answerId,
+                question,
+                questionsFlags,
+                historyQuestions,
+                graderView,
+                showMcqMrqSolution,
+                handleToggleViewHistoryMode,
+              }}
+            />
+            {question.type === questionTypes.Programming && !viewHistory
+              ? renderExplanationPanel(id)
+              : null}
+            {viewHistory ? null : renderQuestionGrading(id)}
+            {viewHistory ? null : renderProgrammingQuestionActions(id)}
+            <Suspense
+              fallback={
+                <>
+                  <br />
```

Comment:
Would be good if this can be removed and margin can be added to the `<div>` below instead.

Reply:
Done!